src/reusedBrowser.ts

```diff
@@ -178,12 +202,18 @@ export class ReusedBrowser implements vscodeTypes.Disposable {
             channel: b.channel,
             title,
           };
-        });
+        }));
       }
 
-      this._pageCountChanged(params.pageCount);
+      this._onPageCountChangedEvent.fire(this.pageCount());
+
+      if (backend === this._testingBackend) {
+        this._testingPageCount = params.pageCount;
+        if (!this._isRunningTests && this._testingPageCount === 0)
```

Comment:
Let's extract a method `_maybeStopTestingBackend()` and call it here, and from `didRunTests()`.